core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/WorkflowWorker.scala

```diff
@@ -0,0 +1,289 @@
+      dataProcessor.breakpoints.foreach { brk =>
+        if (brk.isTriggered)
+          dataProcessor.unhandledFaultedTuples(brk.triggeredTupleId) =
+            new FaultedTuple(brk.triggeredTuple, brk.triggeredTupleId, brk.isInput)
+      }
+      context.parent ! ReportState(WorkerState.LocalBreakpointTriggered)
+    case other =>
+  }
+
+  def getOldWorkerState: WorkerState.Value = {
+    workerStateManager.getCurrentState match {
+      case UnInitialized =>
+        WorkerState.Uninitialized
+      case Ready =>
+        WorkerState.Ready
+      case Running =>
+        WorkerState.Running
+      case Pausing =>
+        WorkerState.Pausing
+      case Paused =>
+        WorkerState.Paused
+      case Completed =>
+        WorkerState.Completed
+      case Recovering =>
+        WorkerState.Running
+    }
+  }
+
+  def reportState(): Unit = context.parent ! ReportState(getOldWorkerState)
+
+  def oldControlMessageHandler: Receive = {
+    case Start =>
+      sender ! Ack
+      if (workerStateManager.getCurrentState != Ready) {
+        logger.info(
+          s"unexpected Start message when worker is in ${workerStateManager.getCurrentState}"
+        )
+      } else if (operator.isInstanceOf[ISourceOperatorExecutor]) {
+        dataProcessor.appendElement(EndMarker())
+        dataProcessor.appendElement(EndOfAllMarker())
+        workerStateManager.transitTo(Running)
+        reportState()
+      } else {
+        logger.info("unexpected Start message for non-source operator!")
+      }
+    case Pause =>
+      promiseManager.execute(PromiseInvocation(null, WorkerPause()))
+      workerStateManager.transitTo(Pausing)
+      reportState()
+    case Resume =>
+      pauseManager.resume()
+      workerStateManager.transitTo(Running)
+      reportState()
+    case AckedWorkerInitialization(recoveryInformation) =>
+      operator.open()
+      workerStateManager.transitTo(Ready)
+      reportState()
+    case QueryState =>
+      reportState()
+    case QueryStatistics =>
+      sender ! ReportStatistics(
+        WorkerStatistics(getOldWorkerState, getInputRowCount(), getOutputRowCount())
+      )
+    case QueryTriggeredBreakpoints =>
+      val toReport = dataProcessor.breakpoints.filter(_.isTriggered)
+      if (toReport.nonEmpty) {
+        toReport.foreach(_.isReported = true)
+        sender ! ReportedTriggeredBreakpoints(toReport)
+      } else {
+        throw WorkflowRuntimeException(
+          WorkflowRuntimeError(
+            "no triggered local breakpoints but worker in triggered breakpoint state",
+            "WorkerBase:allowQueryTriggeredBreakpoints",
+            Map()
+          )
+        )
+      }
+    case QueryBreakpoint(id) =>
+      val toReport = dataProcessor.breakpoints.find(_.id == id)
+      if (toReport.isDefined) {
+        toReport.get.isReported = true
+        context.parent ! ReportedQueriedBreakpoint(toReport.get)
+        context.parent ! ReportState(WorkerState.LocalBreakpointTriggered)
```

Comment:
same question

Reply:
actually I see that we have not done any state check in many of the below messages. eg: ModifyTuple, SkipTuple these shouldnt be allowed in ready and initialized phase.

Reply:
So I don't want to touch the handling logic in this PR. This PR is only about merging processor and generator. We should migrate these handling logics by using promise APIs, and we add the state check when we migrate them.